File: cpp/src/observatory/phase/PreparationPhase.cpp
#include "PreparationPhase.h"
#include <observatory/Benchmark.h>

#include <utility>

namespace Observatory {

PreparationPhase::PreparationPhase(Benchmark &benchmark, uint32_t operationSize, uint32_t operationCount) :
        BenchmarkPhase(benchmark),
        operationSize(operationSize),
        operationCount(operationCount) {}

const char* PreparationPhase::getName() {
    return "PreparationPhase";
}

Status PreparationPhase::execute() {
    return getBenchmark().prepare(operationSize, operationCount);
}

}

File: java/benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/PreparationPhase.java
package de.hhu.bsinfo.observatory.benchmark;

import de.hhu.bsinfo.observatory.benchmark.result.Status;

class PreparationPhase extends BenchmarkPhase {

    private final int operationSize;
    private final int operationCount;

    PreparationPhase(Connection connection, int operationSize, int operationCount) {
        super(connection);

        this.operationSize = operationSize;
        this.operationCount = operationCount;
    }

    @Override
    public Status execute() {
        return getConnection().prepare(operationSize, operationCount);
    }
}
